lib/shipcloud/shipcloud_error.rb

```diff
@@ -0,0 +1,64 @@
+module Shipcloud
+  class ShipcloudError < StandardError
+    attr_reader :errors, :response
+
+    def initialize(response = nil)
+      @response = response
+      @errors = parse_errors
+      error_message = errors.empty? ? response_body : errors
+      super(error_message)
+    end
+
+    # Returns the appropriate Shipcloud::ShipcloudError subclass based
+    # on status code
+    #
+    # @param [HTTPResponse] response HTTP response
+    # @return [Shipcloud::ShipcloudError]
+    def self.from_response(response)
+      response_code = response.code.to_i
+      if error_class = error_class_for(response_code)
+        error_class.new(response)
+      end
+    end
+
+    private
+
+    def self.error_class_for(response_code)
```

Comment:
`private` (on line 24) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.